src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOp.java

```diff
@@ -1156,10 +966,13 @@ public Block.Builder lower(Block.Builder b, OpTransformer opT) {
             }
 
             Block.Builder exit;
-            if (bodies.isEmpty()) {
+            if (bodies().isEmpty()) {
                 exit = b;
             } else {
-                exit = b.block();
+                exit = b.block(resultType());
+                if (result() != null) { // i.e. resultType not void
```

Comment:
All operations in a built model have a result value, even if it's a void value. So this expression will always be true. Use a `this instanceof ...` test?

Reply:
The default value of the `result` field is null

Reply:
It cannot be:
```
    /**
     * Returns the operation's result, otherwise {@code null} if the operation is not assigned to a block.
     *
     * @return the operation's result, or {@code null} if not assigned to a block.
     */
    public final Result result() {
        return result;
    }
```

The field is only null if the operation is not assigned (or bound) to a block, otherwise it has a value even if the type of that value is the void type. An operation is assigned to a block when it is appended via building, so any non-top-level operation in a model must have a result.

Reply:
Yes, you can check if its not equal e.g. `!JavaType.VOID.equals(result())`, but you could also check the `this` instance, since a switch expression yields a non-void value, its arguably clearer in the code.